File: ios/Classes/FlutterUserChannel.swift
import Flutter
import UIKit

public class FlutterUserChannel: NSObject {
    public static let share = FlutterUserChannel()
    static let tokenChannelName = "com.seven.user.flutter_user_channel.token"
    static let otherInfoChannelName = "com.seven.user.flutter_user_channel.other_info"
    static let tokenChannelMethod = "updateWithUserToken"
    static let otherInfoChannelMethod = "transmitOtherInfo"
    private var tokenChannel: FlutterMethodChannel?
    private var otherInfoChannel: FlutterMethodChannel?
    
    
    private override init() {
        super.init()
    }
    public static func updateUserToken(with token: String?, engine: FlutterEngine, callback: @escaping (Any?) -> ()) {
        if share.tokenChannel == nil {
            // 创建渠道
            share.tokenChannel = FlutterMethodChannel(name: tokenChannelName, binaryMessenger: engine.binaryMessenger)
        }
        
        // 通过渠道调用 Flutter 的方法
        let params: Dictionary<String, String> = ["token": token ?? ""]
        share.tokenChannel?.invokeMethod(tokenChannelMethod, arguments: params) { result in
            callback(result)
        }
    }
    /// 传递其他信息
    public static func transmitOtherInfo(with info: [String : Any], engine: FlutterEngine, callback: @escaping (Any?) -> ()) {
        if share.otherInfoChannel == nil {
            // 创建渠道
            share.otherInfoChannel = FlutterMethodChannel(name: otherInfoChannelName, binaryMessenger: engine.binaryMessenger)
        }
        // 通过渠道调用 Flutter 的方法
        let params = ["info": info]
        share.otherInfoChannel?.invokeMethod(otherInfoChannelMethod, arguments: params) { result in
            callback(result)
        }
    }
}

File: android/src/main/kotlin/com/seven/user/flutter_user_channel/FlutterUserChannel.kt
package com.seven.user.flutter_user_channel

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.embedding.engine.FlutterEngine

/** FlutterUserChannelPlugin */
class FlutterUserChannel {
    private var tokenChannel: MethodChannel? = null
    private var otherInfoChannel: MethodChannel? = null
   companion object {
       private var instance: FlutterUserChannel? = null
       fun getInstance(): FlutterUserChannel {
           if (instance == null) {
               synchronized(FlutterUserChannel::class) {
                   if (instance == null) {
                       instance = FlutterUserChannel()
                   }
               }
           }
           return instance!!
       }

       private const val tokenChannelName = "com.seven.user.flutter_user_channel.token"
       private const val otherInfoChannelName = "com.seven.user.flutter_user_channel.other_info"
       private const val tokenChannelMethod = "updateWithUserToken"
       private const val otherInfoChannelMethod = "transmitOtherInfo"


       fun updateUserToken(token: String, engine: FlutterEngine) {
           if (instance?.tokenChannel == null) {
               val messenger = engine.dartExecutor.binaryMessenger
               // 新建一个 Channel 对象
               instance?.tokenChannel = MethodChannel(messenger, tokenChannelName)
           }


           // 调用 Flutter 中的方法
           val params = mapOf(
               "token" to token
           )
           instance?.tokenChannel?.invokeMethod(tokenChannelMethod, params)
       }
       fun transmitOtherInfo(info: Map<String, Any>, engine: FlutterEngine) {
           if (instance?.otherInfoChannel == null) {
               val messenger = engine.dartExecutor.binaryMessenger

               // 新建一个 Channel 对象
               instance?.otherInfoChannel = MethodChannel(messenger, otherInfoChannelName)
           }


           // 调用 Flutter 中的方法
           val params = mapOf(
               "info" to info
           )
           instance?.otherInfoChannel?.invokeMethod(otherInfoChannelMethod, params)
       }
   }
}
